### PR TITLE
333 add consent to uxr contact checkbox to the school onboarding flow

### PR DESCRIPTION
**Pull request by KristinaDudnyk**
## Status

- Closes [#333](https://github.com/RaspberryPiFoundation/digital-editor-issues/issues/333)
- Related to _add issue numbers or delete_

## Points for consideration:

- Security
- Performance

## What's changed?

Created a new migration for school table:
- added new colunm => **creator_agree_to_ux_contact**

## Steps to perform after deploying to production

_If the production environment requires any extra work after this PR has been deployed detail it here. This could be running a Rake task, a migration, or upgrading a Gem. That kind of thing._

### CODE REVIEW

File: db/migrate/20240523143911_add_consent_fields.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 class AddConsentFields < ActiveRecord::Migration[7.0]
   def change
     add_column :schools, :creator_agree_authority, :boolean
-    add_column :schools, :creator_agree_terms_and_conditions, :boolean
+    add_column :schools, :creator_agree_terms_and_conditions, :boolean, default: false
   end
```
[the remark this answers is not in the document]
Comment (KristinaDudnyk) [reply db/migrate/20240523143911_add_consent_fields.rb:5]: Good spotted

File: spec/factories/school.rb [hunk 2]
```diff
@@ -10,6 +10,7 @@
     creator_id { SecureRandom.uuid }
     creator_agree_authority { true }
     creator_agree_terms_and_conditions { true }
+    creator_agree_to_ux_contact { true }
```
Comment (loiswells97) [inline spec/factories/school.rb:13]: I think this isn't needed - we just create the factories with the minimal amount of data present to be valid

Comment (KristinaDudnyk) [reply spec/factories/school.rb:13]: shell I delete this line completely or just change it to false?

Comment (loiswells97) [reply spec/factories/school.rb]: no no what I meant was we don't need this attribute at all in the factory - unless you're validating its presence somewhere?

Comment (KristinaDudnyk) [reply spec/factories/school.rb]: Understood 🫡

**Review by loiswells97 — APPROVED** [review]
Looks great 👍

**Pull request merged**